File: exams/readability/readability.c
/*
This is a program to determine the grade level of books, it scans over the text and calculates these factos:

average amount of letters per word
average aamount of letters per sentence

The formula to calculate the grade:

index = 0.0588 * L - 0.296 * S - 15.8

where L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.
*/

//#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main()
{
    int letters_per_word[] = {};
    int letters_per_sentence[] = {};

    //string book = get_string("Text: ");
    char book[] = "One fish. Two fish. Red fish. Blue fish.";

    int word_array_index = 0;
    int sentence_array_index = 0;

    int word_letters = 0;
    int sentence_letters = 0;

    for(int i = 0; i < strlen(book); i++)
    {
        char c = book[i];

        if(c == ','){
            continue;
        }
        else if(c == ' '){
            if(word_letters > 0){
                letters_per_word[word_array_index] = word_letters;
                word_array_index ++;
                word_letters = 0;
            }else{
                continue;
            }
        }else if(c == '.' | c == '!' | c == '?'){
            letters_per_sentence[sentence_array_index] = sentence_letters;
            letters_per_word[word_array_index] = word_letters;
            sentence_array_index ++;
            word_array_index ++;
            word_letters = 0;
            sentence_letters = 0;
        } else{
            word_letters ++;
            sentence_letters ++;
        }
    }

    for(int i = 0; sizeof(letters_per_word); i++){
        printf("Word %d letter amount: %d \n", i, letters_per_word[i]);
    }

    for(int i = 0; sizeof(letters_per_sentence); i++){
        printf("Sentence %d letter amount: %d \n", i, letters_per_sentence[i]);
    }
}

File: exams/sentimental-readability.py/readability.py
from cs50 import get_string

'''
READABILITY!

grade a text using the us formula to determine the grade level

todo:

get user input
calculate the average amount of letters per 100 words
calculate the average amount of sentences per 100 words
calculate grade and return it
'''


user_input = get_string("Text: ")

grade = 0

avg_let = 0  # average letters
avg_sen = 0  # average sentences


letters = 0
words = 0
sentences = 0
new_word = True

for letter in user_input:
    if letter.isalpha():
        letters += 1
        new_word = False
    elif letter.isspace() and not new_word:
        words += 1
        new_word = True
    elif letter in {".", "!", "?"} and not new_word:
        words += 1
        sentences += 1
        new_word = True
    else:
        continue

avg_let = letters / words * 100
avg_sen = sentences / words * 100

grade = round(0.0588 * avg_let - 0.296 * avg_sen - 15.8)

if grade < 1:
    print("Before Grade 1.")
elif grade > 16:
    print("Grade 16+")
else:
    print(f"Grade: {grade}")
